embassy-nrf/src/spim.rs

```diff
@@ -28,6 +29,40 @@ pub enum Error {
     DMABufferNotInDataMemory,
 }
 
+/// Interface for the SPIM peripheral using EasyDMA to offload the transmission and reception workload.
+///
+/// ## Data locality requirements
+///
+/// On nRF chips, EasyDMA requires the buffers to reside in RAM. However, Rust
+/// slices will not always do so. Take the following example:
+///
+/// ```no_run
+/// // As we pass a slice to the function whose contents will not ever change,
+/// // the compiler writes it into the flash and thus the pointer to it will
+/// // reference static memory. Since EasyDMA requires slices to reside in RAM,
+/// // this function call will fail.
+/// let result = spim.write_from_ram(&[1, 2, 3]);
+/// assert_eq!(result, Error::DMABufferNotInDataMemory);
```

Comment:
This is `Err(Error::DMABufferNotInDataMemory)`

Reply:
Good catch. I'll add another commit in the following days that moves the docs up to the module level — that should then include a fix for this.